environments/environments.py

```diff
@@ -0,0 +1,65 @@
+from abc import ABC, abstractmethod
+import numpy as np
+
+
+class Environment(ABC):
+
+    @abstractmethod
+    def set(self, p, q):
+        """Sets initial conditions for physical system
+
+        Args:
+            p (float): generalized momentum
+            q (float): generalized position
+
+        Raises:
+            NotImplementedError: Class instantiation has no implementation
+        """
+        raise NotImplementedError
```

Comment:
If I'm not mistaken, a class that extends `ABC` throws an error if it's instantiated without implementation of methods decorated with `@abstractmethod`. Thus, it should not be necessary to raise the `NotImplementedError` in all these methods:

```suggestion
        """
        pass
```

Reply:
Moreover, if someone tries to call `super().set()` (which is something kinda common) in the implementation of the derived class, then an error is raised

Reply:
So then, I remove all the raise NotImplementedError in the abstract class?

Reply:
Yes, you can substitute them with just `pass`.